### ISSUE TITLE
Shipping Class are not sync for Product Variations 

### ISSUE THREAD

**Issue by decarvalhoaa** [hyyan/woo-poly-integration#112]
Steps to reproduce:

1. For a Variable Product, change the Shipping Class of a Variation to other value than "Same as parent"
2. Click "Save changes" or "Update" to update the entire Product.
3. Open a product translation and check the Shipping Class of the matching Variation for which the shipping class was change in step 1.

This is can also be reproduce for both new and existing Variable Products.

**Comment by Jon007** [comment]
oh yes, it doesn't attempt to sync variation shipping class

At the product level, Meta.php has a syncShippingClass() method to cope with the fact that shipping class is a term not a meta so not included in the meta copying.

variation.php has no similar syncShippingClass() to implement this for variations, but looks like it could be implemented along similar lines, though I didn't quite succeed yet.

Still, does this all change with woo 2.7 anyway?

**Comment by Jon007** [comment]
FIX PROPOSED:

How about this in Variation.php:

1) add a function to copy shipping class
		
```
    /**
     * Sync Product Shipping Class.
     * 
     * Shipping Class translation is not supported after WooCommerce 2.6
     * but it is still implemented by WooCommerce as a taxonomy. Therefore,
     * Polylang will not copy the Shipping Class meta.
     *
     * @param int $from product variation ID
     * @param int $to   product variation ID
     */
    public function syncShippingClass($from, $to)
    {
			// Product edit - update shipping class of all product translations
			$variation_from = wc_get_product($from);
            
			if ($variation_from) {           
				$shipping_class = $variation_from->get_shipping_class();
				if ($shipping_class){
					$shipping_terms = get_term_by( 'slug', $shipping_class, 'product_shipping_class' );
					if ($shipping_terms) {
						wp_set_post_terms( $to, array( $shipping_terms->term_id ), 'product_shipping_class' );
					}
				}
			}    
		}

```
2) call this function when copying attributes, simply adding the line to copyVariationMetas:

```
    protected function copyVariationMetas($from, $to)
    {
				$this->syncShippingClass($from, $to);

```

I'm not entirely familiar enough with the code to know if this is the best solution, but it seems reasonably straightforward, and in my tests so far, it appears to work.

**Comment by Jon007** [comment]
Note that this fix applies when the Product is saved using the Update button on the product.

It is still possible to cause the variations to become out of sync by pressing "Save changes" on the Variations panel, which does an ajax call which saves the variations on the main language product without doing a synchronization.

**Comment by Jon007** [comment]
The remaining issue about the Ajax call not synchronizing translations is generic and explains various other issue reported, will track this separately on #146

Fixed included in pull request 0.30RC1, the code is available at:
https://github.com/Jon007/woo-poly-integration

This should be treated as beta software until we have completed the testing, merge and release.

It would be great if you can test it!!!
If you find any issues with the RC, please report as separate issues, I will fix asap.
